cluster.go

```diff
@@ -111,6 +112,15 @@ func ClusterWithTrace(ct trace.ClusterTrace) ClusterOpt {
 	}
 }
 
+// ClusterWithInitSyncSilent tells the Cluster whether it should ignore
+// errors of creating pool for addrs during the initialization
+// if pool for all addrs failed to be created, it will still return errors
+func ClusterWithInitSyncSilent(initSyncSilent bool) ClusterOpt {
```

Comment:
I'd like to rename this to `ClusterOnInitAllowUnavailable`, to be more specific about what it's doing, and change the comment to:

```
ClusterOnInitAllowUnavailable tells NewCluster to succeed and not return an error as long as at least one redis instance in the cluster can be successfully connected to.
```